src/pypowsybl_jupyter/networkexplorer.py

```diff
@@ -206,20 +214,83 @@ def on_selected_history(d):
 
     history.observe(on_selected_history, names='value')
 
-    def update_explorer():
+    def compute_nad_data(el, depth=0):
+        if el is not None:
+            nad_data=network.get_network_area_diagram(voltage_level_ids=el, 
+                                                      depth=depth, high_nominal_voltage_bound=high_nominal_voltage_bound, 
+                                                      low_nominal_voltage_bound=low_nominal_voltage_bound, nad_parameters=npars)
+        else:
+            nad_data=EMPTY_SVG
+        return nad_data
+
+    current_nad_data = compute_nad_data(None)
+
+    def update_nad_widget(new_diagram_data, enable_callbacks=True, grayout=False):
+        nonlocal nad_widget
+        if nad_widget==None:
+            nad_widget=display_nad(new_diagram_data, enable_callbacks=enable_callbacks, grayout=grayout)
+            nad_widget.on_select_node(lambda event : go_to_vl_from_nad(event))
+        else:
+            update_nad(nad_widget,new_diagram_data, enable_callbacks=enable_callbacks, grayout=grayout)
+
+    in_progress_widget=widgets.HTML(value=PROGRESS_EMPTY_SVG, 
+                                    layout=widgets.Layout(width='30', justify_content='flex-end', margin='0px 20px 0px 0px'))
+
+    def enable_in_progress():
+        in_progress_widget.value=PROGRESS_BAR_SVG
+        found.disabled=True
+        history.disabled=True
+        nadslider.disabled=True
+        vl_input.disabled=True
+        display(vl_input)
+
+    def disable_in_progress():
+        found.disabled=False
+        history.disabled=False
+        nadslider.disabled=False
+        vl_input.disabled=False
+        in_progress_widget.value=PROGRESS_EMPTY_SVG
+
+    def update_nad_diagram(el):
+        nonlocal current_nad_data, nad_displayed_vl_id
+        if nad_widget != None:
+            enable_in_progress()
+            update_nad_widget(current_nad_data, enable_callbacks=False, grayout=True)
+            display(nad_widget)
+            update_sld_widget(current_sld_data, True, enable_callbacks=False)
+            display(sld_widget)
+            if map_widget != None:
+                map_widget.set_enable_callbacks(False)
+                display(map_widget)
+        try:
+            current_nad_data=compute_nad_data(el, selected_depth)
+            update_nad_widget(current_nad_data, enable_callbacks=True, grayout=False)
+            nad_displayed_vl_id=el
+        finally:
+            update_sld_widget(current_sld_data, True, enable_callbacks=True)
+            if map_widget != None:
+                map_widget.set_enable_callbacks(True)
+
+            disable_in_progress()
+    
+    def update_explorer(force_update=False):
         sel=sel_ctx.get_selected()
-        update_nad_diagram(sel)
+        if force_update or tabs_diagrams.selected_index==NAD_TAB_INDEX:
+            update_nad_diagram(sel)
```

Comment:
In `update_nad_diagram`, there's some sld update, because of disabling the callbacks, the update being made with the same svg content. So we're calling 3 times the sld update, but it's not slowing down the sld update because the string is unchanged when disabling/enabling the callbacks?